corehq/apps/users/util.py

```diff
@@ -109,6 +109,31 @@ def username_to_user_id(username):
     return user._id
 
 
+def cached_location_id_to_name(location_id):
```

Comment:
@esoergel Do you have any recommendations for location caching? I'd think we might have existing code that would be useful here.

Reply:
how about using quickcache instead of managing redis key?
also, should this go in the locations app.

Reply:
I was also going to suggest quickcache over manual key management. @Robert-Costello have you seen this docs page on it? https://commcare-hq.readthedocs.io/caching_and_memoization.html

Couple higher-level questions on this though - what is the use-case for this being cached?  Is it primarily to avoid multiple lookups of the same ID in a single request, or to store them across requests?

If the target is a single request - I'd probably use an in-memory cache object along the lines of
https://github.com/dimagi/commcare-hq/blob/49e3b311c061704114c9b008b09de52afe1b5705/corehq/apps/fixtures/upload/location_cache.py
If access is scoped to a single view class, I'd just use the `@memoized` decorator on a class method

If this is across requests, then quickcache is a better option.  However, I'm not sure that use-case would be a great cache target unless we have evidence to suggest it is.  Fetching a single object by ID is a very fast postgres query, so I think we'd need a decent argument to warrant adding it to the cache - say if the same small set of IDs were fetched a lot.

A big downside to caching like this is that we'd need to consider invalidation, which as they say is one of the two hard problems in computer science:
* cache invalidation
* naming things
* off-by-one errors

Reply:
Looking at this after having reviewed the whole PR, I see you were mirroring the structure of `cached_user_id_to_username`.  One important difference there is that usernames can't change, so there's no need to invalidate that cache.

One easy way to memoize things like this ID lookup would be to make the helper functions for `UserHistoryReport` to be methods of that report - then you can just make a method like this:
```python
    @memoized
    def _get_location_name(self, location_id):
        ...
```
And it will only do the lookup once per ID.

Another thought - you might want to use `location.display_name` (which includes the location type), or `location.get_path_display`, though the latter does a more expensive query.